coiled_calib.py

```diff
@@ -24,74 +23,71 @@ def calibrate(self, calib_pars=None, load=False, tidyup=True, **kwargs):
             verbose (bool): whether to print output from each trial
             kwargs (dict): if supplied, overwrite stored run_args (n_trials, n_workers, etc.)
         """
+        import optuna as op
+
         # Load and validate calibration parameters
         if calib_pars is not None:
             self.calib_pars = calib_pars
         self.run_args.update(kwargs) # Update optuna settings
 
-        # Run the optimization
-        t0 = sc.tic()
-        self.study = self.make_study()
-
-        #self.run_workers() # <-- This is the line from the base class to run locally using multiprocessing
-        ################################
-        cluster = coiled.Cluster(
+        with coiled.Cluster(
             n_workers=10,
-            name='StarsimCalibrationOnCoiled'
-        )
-
-        # backend_storage = op.storages.InMemoryStorage() --> not sure this is needed
-        dask_storage = op.integration.DaskStorage(storage=None,
-                                                  client=cluster.get_client())
-
-        client = cluster.get_client()
-        futures = [
-            client.submit(
-                self.study.optimize,
-                self.run_trial,
-                n_trials=self.run_args.n_trials,
-                pure=False
-            )
-            for _ in range(self.run_args.n_workers)
-        ]
-
-        wait(futures)
-        #################################
-
-        study = op.load_study(storage=dask_storage, study_name=self.run_args.study_name, sampler=self.run_args.sampler)
-        self.best_pars = sc.objdict(study.best_params)
-        self.elapsed = sc.toc(t0, output=True)
-
-        self.sim_results = []
-        if load:
-            if self.verbose: print('Loading saved results...')
-            for trial in study.trials:
-                n = trial.number
-                try:
-                    filename = self.tmp_filename % trial.number
-                    results = sc.load(filename)
-                    self.sim_results.append(results)
-                    if tidyup:
+            name='StarsimCalibrationOnCoiled',
+        ) as cluster:
```

Comment:
@daniel-klein here I'm using a context manager for always shutting things down. There are also explicit 

```python
cluster.close()
client.close()
```

methods if you prefer those